core/src/main/scala/org/apache/spark/sql/collection/Utils.scala

```diff
@@ -368,6 +368,52 @@ object Utils {
     k
   }
 
+  /**
+   * Utility function to return a metadata for a StructField of StringType, to ensure that the
+   * field is stored (and rendered) as VARCHAR by SnappyStore.
+   * @param md
+   * @param size
+   * @return
+   */
+  def varcharMetadata(md: Metadata = Metadata.empty, size: Int = Constant.MAX_VARCHAR_SIZE):
```

Comment:
I think we should make these methods Java friendly too. So I will suggest overloads instead of default arguments:

```scala
def varcharMetadata(): Metadata = varcharMetadata(Constant.MAX_VARCHAR_SIZE)
def varcharMetadata(size: Int): Metadata = varcharMetadata(size, Metadata.empty)
def varcharMetadata(size: Int, md: Metadata): Metadata = ...
```